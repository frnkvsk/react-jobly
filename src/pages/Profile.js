import React, {useState, useContext} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Container, FormHelperText, } from '@material-ui/core';
import OutlinedInput from '@material-ui/core/OutlinedInput';
import { Button, Box } from '@material-ui/core';
import { AuthContext } from '../context/AuthContext';
import { useFormInput } from '../hooks/useFormInput';
import { patchUserInfo } from '../api/JoblyApi';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    maxWidth: '500px',
    padding: '30px',
    height: '100vh',
  },
  form: {
    width: '100%',
    backgroundColor: '#ffffff',
  },
  label: {
    fontSize: '18px',
    fontWeight: 'bold',
    width: '90%',
    margin: '10px 20px 0 20px',
  },
  input: {
    width: '90%',
    margin: '0 20px 0 20px',
  },
  button: {
    display: 'flex',
    justifyContent: 'flex-end',
    margin: '10px 20px 15px',
  },
  success: {
    color: '#4caf50',
    fontSize: '24px',
  },
  err: {
    color: '#ff1744',
    fontSize: '24px',
  }

}));

export default function Profile() {
  const auth = useContext(AuthContext);
  const user = auth.authState.userInfo;
  const classes = useStyles();
  const username = useFormInput(user.username);
  const password = useFormInput("");
  const first_name = useFormInput(user.first_name);
  const last_name = useFormInput(user.last_name);
  const email = useFormInput(user.email);
  const photo_url = useFormInput(user.photo_url);  
  const [successMessage, setSuccessMessage] = useState(false);
  const [errorMessage, setErrorMessage] = useState(false);
  const handleClick = async () => {
    try {
      const userInfo = {
        jobs: auth.authState.userInfo.jobs,
        username: username.value,
        first_name: first_name.value,
        last_name: last_name.value,        
        email: email.value,
        photo_url: photo_url.value,
      }
      // set context state and save it to localstorage
      auth.setAuthState({
        token: auth.authState.token,
        userInfo: userInfo,
      });
      // persist data to the database
      delete userInfo.username;
      delete userInfo.jobs;
      userInfo.password = password.value;
      if(Object.values(userInfo).every(e => e.length)) {
        await patchUserInfo(auth.authState.token, username.value,userInfo);
        setErrorMessage(false);
        setSuccessMessage(true);
      } else {
        setSuccessMessage(false);
        setErrorMessage(true);
      }
      
    } catch (error) {
      console.error('Profile err',error);
    }
    
  }

  return (
    
    <Container className={classes.root}>
      <h1>Profile</h1>  
      <Box className={classes.success} 
        component="span" 
        display={successMessage ? 'block' : 'none'}
        >
          Success: Profile updated
      </Box>    
      <Box className={classes.err} 
        component="span" 
        display={errorMessage ? 'block' : 'none'}
        >
          Error: Inputs incomplete
      </Box>    
      <form className={classes.form}>      
        <FormHelperText className={classes.label} >Username</FormHelperText>
        <FormHelperText className={classes.label}>{username.value}</FormHelperText>

        <FormHelperText className={classes.label}>Confirm Password</FormHelperText>
        <OutlinedInput required={true} className={classes.input} name="password" type="password" variant="outlined" {...password} />

        <FormHelperText className={classes.label}>First Name</FormHelperText>
        <OutlinedInput className={classes.input} name="first_name" variant="outlined" {...first_name}/>

        <FormHelperText className={classes.label}>Last Name</FormHelperText>
        <OutlinedInput className={classes.input} name="last_name" variant="outlined" {...last_name}/>

        <FormHelperText className={classes.label}>Email</FormHelperText>
        <OutlinedInput className={classes.input} name="email" variant="outlined" {...email}/>

        <FormHelperText className={classes.label}>Photo URL</FormHelperText>
        <OutlinedInput className={classes.input} name="photo_url" variant="outlined" {...photo_url}/>
        <div className={classes.button}>
          <Button onClick={handleClick} variant="contained" color="primary">Update Profile</Button>
        </div>          
      </form>
      
    </Container>
  );
}