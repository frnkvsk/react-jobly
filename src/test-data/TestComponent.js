// import React, { useContext } from 'react';
// import { AuthContext } from '../context/AuthContext';

// const TestComponent = () => {
//   const auth = useContext(AuthContext);
//   const setAuth2 = async () =>
//     await auth.setAuthState({
//       token: 'testtoken',
//       userInfo: {
//       username: 'testuser',
//       first_name: 'testfirst',
//       last_name: 'testlast',
//       email: 'test@test.com',
//       photo_url: null
//     }});
//   (async () => await setAuth2())();
//   console.log('-------auth.authState',auth.authState)
//   return (
//     <div>
//       <div>{auth.authState.userInfo.username}</div>
//       <div>{auth.authState.userInfo.first_name}</div>
//       <div>{auth.authState.userInfo.last_name}</div>
//       <div>{auth.authState.userInfo.email}</div>
//       <div>{auth.authState.userInfo.photo_url}</div>
//     </div>
//   )
// }
// export default TestComponent;

