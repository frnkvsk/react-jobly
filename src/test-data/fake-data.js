const testData = {
  jobs: [
    {
    company_handle: "test-company-a",
    equity: 0.115,
    id: 1,
    salary: 11111,
    state: null,
    title: "Editor, magazine features",
    },
      {
      company_handle: "test-company-b",
      equity: 0.22,
      id: 2,
      salary: 22222,
      state: null,
      title: "Coder",
    },
      {
      company_handle: "test-company-c",
      equity: 0.33,
      id: 3,
      salary: 33333,
      state: null,
      title: "Front-end Dev",
    },
  ],
  companies: [
    {
      description: "Test description 1",
      handle: "test-company-1",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 1",
    },
    {
      description: "Test description 2",
      handle: "test-company-2",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 2",
    },
    {
      description: "Test description 3",
      handle: "test-company-3",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 3",
    },
  ],
}

export { testData }