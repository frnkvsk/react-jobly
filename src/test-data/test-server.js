import { rest } from 'msw';
import { setupServer } from 'msw/node';

let server;

const setServer = async (url, status, message='') => {
  if(server) {
    server.close();
    server.resetHandlers();
  }
  server = await setupServer(
    rest.get(url, (req, res, ctx) => {
      return res(
        ctx.status(status),
        ctx.json(message)
      )
    })
  );
}
  

beforeAll(() => server ? server.listen() : null);
afterAll(() => server ? server.close() : null);
afterEach(() => server ? server.resetHandlers() : null);

export { setServer, server, rest }
