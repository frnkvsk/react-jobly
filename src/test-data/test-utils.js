// test-utils.js
import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { SearchProvider } from '../context/SearchContext';
import { AuthProvider } from '../context/AuthContext';
import { PageCountProvider } from '../context/PageCountContext';

const AllTheProviders = (page) => {
  return render(
    <BrowserRouter>
      <AuthProvider>
        <SearchProvider>
          <PageCountProvider>
            {page}
          </PageCountProvider>
        </SearchProvider>
      </AuthProvider> 
    </BrowserRouter>
    
  );
}

// re-export everything
export * from '@testing-library/react';

// override render method
export { AllTheProviders as render }
