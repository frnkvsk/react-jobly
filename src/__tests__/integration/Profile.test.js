import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../test-data/test-utils';
import Profile from '../../pages/Profile';

// smoke and appearance tests
test('renders Profile ', async() => {
  render(<Profile />);
  
  expect(screen.getAllByText(/profile/i)[0]).toBeInTheDocument();
  
  expect(screen.getByText(/confirm password/i)).toBeInTheDocument();

  expect(screen.getByRole('button', {name: /update profile/i})).toBeInTheDocument();
});

// snapshot test
// test('matches snapshot', () => {
//   const { asFragment } = render(<Profile />);
//   expect(asFragment()).toMatchSnapshot();
// });