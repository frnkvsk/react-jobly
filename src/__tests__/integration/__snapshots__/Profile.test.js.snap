// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-root-14 MuiContainer-maxWidthLg"
  >
    <h1>
      Profile
    </h1>
    <span
      class="MuiBox-root MuiBox-root-21 makeStyles-success-19"
    >
      Success: Profile updated
    </span>
    <span
      class="MuiBox-root MuiBox-root-22 makeStyles-err-20"
    >
      Error: Inputs incomplete
    </span>
    <form
      class="makeStyles-form-15"
    >
      <p
        class="MuiFormHelperText-root makeStyles-label-16"
      >
        Username
      </p>
      <p
        class="MuiFormHelperText-root makeStyles-label-16"
      />
      <p
        class="MuiFormHelperText-root makeStyles-label-16"
      >
        Confirm Password
      </p>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-17"
        required=""
        variant="outlined"
      >
        <input
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="password"
          required=""
          type="password"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-23 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-24"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-label-16"
      >
        First Name
      </p>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-17"
        variant="outlined"
      >
        <input
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="first_name"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-23 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-24"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-label-16"
      >
        Last Name
      </p>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-17"
        variant="outlined"
      >
        <input
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="last_name"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-23 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-24"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-label-16"
      >
        Email
      </p>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-17"
        variant="outlined"
      >
        <input
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="email"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-23 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-24"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root makeStyles-label-16"
      >
        Photo URL
      </p>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root makeStyles-input-17"
        variant="outlined"
      >
        <input
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="photo_url"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-23 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-24"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
      <div
        class="makeStyles-button-18"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Update Profile
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
